<launch>

    <!-- Loading Arguments -->
    <arg name="planner_type"        default="rrt_star"/>
    <arg name="scenario_number"        default="1"/>
    <arg name="num_pos_initial"         default="1"/>
    <arg name="pos_uav_above_ugv"     default="1.0"/> #meters UAV above UGV
    <arg name="rviz"    default="true"/> 
    <arg name="viz_file"    default="marsupial"/>
    <arg name="is_coupled"    default="false"/>
    <arg name="sample_mode"    default="1"/>
    <arg name="write_data_files" default="true"/>
    
    <arg name="pos_reel_x"    default="0.0"/> #position reel on UGV 
    <arg name="pos_reel_y"    default="0.0"/> #position reel on UGV
    <arg name="pos_reel_z"    default="0.5"/> #position reel on UGV

    <arg name="map_file" default="stage_$(arg scenario_number)"/>

    <!-- Publishing UGV and UAV tf -->
    <node pkg="marsupial_optimizer" type="optimizer_publisher_tf_node" name="optimizer_publisher_tf_node" output="screen">
        <rosparam command="load" file="$(find marsupial_optimizer)/cfg/stage_$(arg scenario_number).yaml" />
        <param name="num_pos_initial" value="$(arg num_pos_initial)"/>
        <param name="pos_uav_above_ugv" value="$(arg pos_uav_above_ugv)"/>
        <param name="uav_base_frame" value="uav_base_link"/>
        <param name="ugv_base_frame" value="ugv_base_link"/>
        <param name="reel_base_frame" value="reel_base_link"/>
    </node>

    <!-- Publishing static tf -->
    <node pkg="tf" type="static_transform_publisher" name="map_tf_node" args="0 0 0 0 0 0  world /map 10"/> # x y z y p r
    <node pkg="tf" type="static_transform_publisher" name="reel_base_link_tf_node" args="$(arg pos_reel_x) $(arg pos_reel_y) $(arg pos_reel_z) -0.0 0.0 0.0  ugv_base_link reel_base_link 10"/> # x y z y p r

    <!-- Launch Octomap of Scenario -->
    <include file="$(find marsupial_optimizer)/launch/octomap_server.launch">
        <arg name="map_file"  value="stage_$(arg scenario_number)"/>    
    </include>

    <!-- Launch  region_growing-->
    <!-- <node pkg="timed_roslaunch" type="timed_roslaunch.sh" name="region_growing_time" output="screen" 
    args="0 rrt_planners region_growing.launch  "/> -->
    <include file="$(find rrt_planners)/launch/region_growing.launch">
    </include> 

    <!-- Launch Optimizer and Global Planner-->
    <include file="$(find rrt_planners)/launch/random_global_planner.launch">
        <arg name="coupled" value="$(arg is_coupled)"/> 
        <arg name="scenario_number" value="$(arg scenario_number)"/> 
        <arg name="num_pos_initial" value="$(arg num_pos_initial)" /> 
        <arg name="sample_mode_rrt" value="$(arg sample_mode)" /> 
        <arg name="planner_type" value="$(arg planner_type)" /> 
        <arg name="write_data_files" value="$(arg write_data_files)" /> 
    </include>    

    <!-- RViz -->
    <!-- <node pkg="rviz" name="rviz" type="rviz" args="-d $(find marsupial_optimizer)/viz/$(arg viz_file).rviz" if="$(arg rviz)"/> -->

</launch>